### PR TITLE
Fix the gelu backward to avoid nan

### PR DESCRIPTION
**Pull request by kuke**

### CODE REVIEW

**Review by iclementine — DISMISSED** [review]
It works now, changes listed
1. removing division to avoid nan;
2. remove unnecessary eval() control;
3. remove unnecessary different inplace() than that from the base class.

File: paddle/fluid/operators/activation_op.h [hunk 1]
```diff
@@ -309,15 +309,17 @@ struct GeluFunctor : public BaseActivationFunctor<T> {
...
             typename dX>
   void operator()(Device d, X x, Out out, dOut dout, dX dx) const {
-    auto temp = (static_cast<T>(0.5 * M_2_SQRTPI * M_SQRT1_2) * x *
-                 ((-static_cast<T>(0.5) * x.square()).exp()))
-                    .template cast<T>()
-                    .eval();
-    dx.device(d) = dout * (out / x + temp);
+    auto first = static_cast<T>(0.5) *
+                 (static_cast<T>(1) +
+                  ((x * static_cast<T>(M_SQRT1_2)).erf()).template cast<T>());
```
Comment (chengduoZH) [inline paddle/fluid/operators/activation_op.h]: It seems that `.template cast<T>()` is unnecessary.

Comment (kuke) [reply paddle/fluid/operators/activation_op.h]: Done

**Review by chengduoZH — APPROVED** [review]
Excellent

**Pull request merged**